main/src/modules/Assembly.scala

```diff
@@ -77,52 +70,39 @@ object Assembly {
     }
   }
 
-  private def classpathIterator(inputPaths: Agg[os.Path]): Generator[AssemblyEntry] = {
-    Generator.from(inputPaths)
+  private def classpathIterator(inputPaths: Agg[os.Path]): Agg[(String, InputStream)] =
+    inputPaths
       .filter(os.exists)
-      .flatMap {
-        p =>
-          if (os.isFile(p)) {
-            val jf = new JarFile(p.toIO)
-            Generator.from(
-              for(entry <- jf.entries().asScala if !entry.isDirectory)
-                yield JarFileEntry(entry.getName, () => jf.getInputStream(entry))
-            )
-          }
-          else {
-            os.walk.stream(p)
-              .filter(os.isFile)
-              .map(sub => PathEntry(sub.relativeTo(p).toString, sub))
-          }
+      .flatMap { path =>
+        if (os.isFile(path)) {
+          val jarFile = new JarFile(path.toIO)
+          jarFile
+            .entries()
+            .asScala
+            .filterNot(_.isDirectory)
+            .map(entry => entry.getName -> jarFile.getInputStream(entry))
```

Comment:
jarjar-abram is looking for a `Seq[(String, InputStream)]`.

Reply:
Oh, I see. Maybe then, `Shader.shadeInputStreams` in jarjar-abrams could be changed to something like
```scala
def processByteCode(
    rules: Seq[ShadeRule],
    verbose: Boolean
): (String, Array[Byte]) => Option[(String, Array[Byte])]
```
This method could be used to process the `InputStream`s one by one.

Reply:
I've done `(String, Array[Byte]) => Option[(String, Array[Byte])]` in https://github.com/eed3si9n/jarjar-abrams/pull/6
I monkey patched in this PR to see how it plays. What's your thought on this?

Reply:
I pushed again with `Array[Byte]` let me know what you think.

Reply:
After thinking about this, I feel like I prefer the `InputStream` over `Array[Byte]` because at least with an `InputStream` we are not loading all the jars in memory if there is no rules. And also in the future we can think of jarjar-abram or another one that could process the shading on the fly inside an `InputStream`. Maybe?